include/boost/url/impl/url_base.hpp

```diff
@@ -22,194 +22,48 @@ segments_type(
 {
 }
 
-url_view::
-params_type::
-params_type(
-    url_base const& v) noexcept
-    : s_(v.s_)
-    , pt_(&v.pt_)
+bool
+url_base::
+segments_type::
+iterator::
+operator==(
+    iterator const& other) const noexcept
```

Comment:
iterators are passed by value

Reply:
You've done this in many places, you want me to change it?

Reply:
I changed it. You were only doing this in op== and op!=.

Reply:
If I passed by `const&`, it was wrong